features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt

```diff
@@ -7839,6 +7858,433 @@ internal class RumViewScopeTest {
+        expectedAttributes.putAll(fakeNewGlobalAttributes)
+        whenever(rumMonitor.mockInstance.getAttributes()) doReturnConsecutively listOf(
+            // one for initialization
+            fakeGlobalAttributes,
+            // second for event handling
+            fakeNewGlobalAttributes
+        )
+
+        testedScope = RumViewScope(
+            mockParentScope,
+            rumMonitor.mockSdkCore,
+            fakeKey,
+            fakeEventTime,
+            fakeAttributes,
+            mockViewChangedListener,
+            mockResolver,
+            mockCpuVitalMonitor,
+            mockMemoryVitalMonitor,
+            mockFrameRateVitalMonitor,
+            featuresContextResolver = mockFeaturesContextResolver,
+            trackFrustrations = fakeTrackFrustrations,
+            sampleRate = fakeSampleRate
+        )
+        mockSessionReplayContext(testedScope)
+
+        // When
+        val result = testedScope.handleEvent(
+            RumRawEvent.StopView(fakeKey, emptyMap()),
+            mockWriter
+        )
+
+        // Then
+        argumentCaptor<ViewEvent> {
+            verify(mockWriter).write(eq(mockEventBatchWriter), capture())
+            assertThat(lastValue).containsExactlyContextAttributes(expectedAttributes)
+        }
+        verifyNoMoreInteractions(mockWriter)
+        assertThat(result).isNull()
+    }
+
+    @Test
+    fun `𝕄 not update the global attributes 𝕎 handleEvent(StartView)`(
+        forge: Forge
+    ) {
+        // Given
+        val fakeGlobalAttributes = forge.aFilteredMap(excludedKeys = fakeAttributes.keys) {
+            anHexadecimalString() to anAsciiString()
+        }
+        val fakeNewGlobalAttributes = forge.aFilteredMap(excludedKeys = fakeAttributes.keys) {
+            anHexadecimalString() to anAsciiString()
+        }
+        val expectedAttributes = mutableMapOf<String, Any?>()
+        expectedAttributes.putAll(fakeAttributes)
+        expectedAttributes.putAll(fakeGlobalAttributes)
+        whenever(rumMonitor.mockInstance.getAttributes()) doReturnConsecutively listOf(
+            // one for initialization
+            fakeGlobalAttributes,
+            // second for event handling
+            fakeNewGlobalAttributes
+        )
+
+        testedScope = RumViewScope(
+            mockParentScope,
+            rumMonitor.mockSdkCore,
+            fakeKey,
+            fakeEventTime,
+            fakeAttributes,
+            mockViewChangedListener,
+            mockResolver,
+            mockCpuVitalMonitor,
+            mockMemoryVitalMonitor,
+            mockFrameRateVitalMonitor,
+            featuresContextResolver = mockFeaturesContextResolver,
+            trackFrustrations = fakeTrackFrustrations,
+            sampleRate = fakeSampleRate
+        )
+        mockSessionReplayContext(testedScope)
+
+        // When
+        val result = testedScope.handleEvent(
+            RumRawEvent.StartView(fakeKey, emptyMap()),
```

Comment:
in theory we cannot receive `StartView` event with the same key as the active scope already has (it should be the key inherited from another view)

Reply:
yes that's correct, let me change that. For me was more important the attributes update trigger but yes to be 100% correct a different key here would be needed.